scheduler-server/src/test/java/gov/nasa/jpl/aerie/scheduler/server/services/SchedulingIntegrationTests.java

```diff
@@ -0,0 +1,371 @@
+          .add(activity);
+    }
+    return result;
+  }
+
+  private static Plan makePlan(final Iterable<PlannedActivityInstance> activities, final Problem problem) {
+    final var initialPlan = new PlanInMemory();
+
+    for (final var activity : activities) {
+      final var activityType = problem.getActivityType(activity.type());
+      final var activityInstance = new ActivityInstance(activityType, Duration.ZERO, Duration.ZERO);
+      for (final var entry : activity.args().entrySet()) {
+        activityInstance.addArgument(entry.getKey(), entry.getValue());
+      }
+      if (activityType.getDurationType() instanceof DurationType.Controllable a) {
+        final var duration = Optional.ofNullable(activity.args.get(a.parameterName()));
+        duration.ifPresent(serializedDuration -> activityInstance.setDuration(new DurationValueMapper().deserializeValue(serializedDuration).getSuccessOrThrow()));
+      }
+      initialPlan.add(activityInstance);
+    }
+    return initialPlan;
+  }
+
+  record PlannedActivityInstance(String type, Map<String, SerializedValue> args, Duration startTime) {}
+
+  private SchedulingRunResults runSchedulerOnBanananation(List<PlannedActivityInstance> plannedActivities, final Iterable<String> goals)
+  throws SchedulingDSLCompilationService.SchedulingDSLCompilationException
+  {
+    this.merlinService.setInitialPlan(plannedActivities);
+    final var planId = new PlanId(1L);
+    final var goalsByPriority = new ArrayList<GoalRecord>();
+    var goalId = 0L;
+    for (final var goal : goals) {
+      final var goalSpecifier = schedulingDSLCompiler.compileSchedulingGoalDSL(planId, goal, "");
+      goalsByPriority.add(new GoalRecord(new GoalId(goalId++), goalSpecifier));
+    }
+    final var specificationService = new MockSpecificationService(Map.of(new SpecificationId(1L), new Specification(
+        planId,
+        1L,
+        goalsByPriority,
+        Timestamp.fromString("2021-001T00:00:00"),
+        Timestamp.fromString("2021-005T00:00:00"),
+        Map.of())));
+    final var agent = new SynchronousSchedulerAgent(specificationService, this.merlinService, this.banananationLibPath, Path.of(""), PlanOutputMode.UpdateInputPlanWithNewActivities);
+    // Scheduling Goals -> Scheduling Specification
+    final var writer = new MockResultsProtocolWriter();
+    agent.schedule(new ScheduleRequest(new SpecificationId(1L), $ -> RevisionData.MatchResult.success()), writer);
+    assertEquals(1, writer.results.size());
+    final var result = writer.results.get(0);
+    assert result instanceof MockResultsProtocolWriter.Result.Success;
+    return new SchedulingRunResults(((MockResultsProtocolWriter.Result.Success) result).results(), this.merlinService.updatedPlan);
+  }
+
+  record SchedulingRunResults(ScheduleResults scheduleResults, Collection<PlannedActivityInstance> updatedPlan) {}
+
+  class MockResultsProtocolWriter implements ResultsProtocol.WriterRole {
```

Comment:
Can you separate the mock classes into their own files like is done in merlin server? It'll make this file a little faster to understand.

Reply:
Good idea - I went ahead and split them into separate files.